python/sglang/lang/chat_template.py

```diff
@@ -42,7 +43,8 @@ def get_prompt(self, messages: List[Dict]) -> str:
         prompt = ""
         for i, message in enumerate(messages):
             role, content = message["role"], message["content"]
-            if role == "system" and content is None:
+            # if use default system like qwen
+            if role == "system" and (content is None or self.use_default_system):
```

Comment:
why is this needed? To use the default system prompt, you can just leave the content empty.

Reply:
@for-just-we any updates? We want to allow users to override the system prompt so i think `use_default_system` is not required.

Reply:
> @for-just-we any updates? We want to allow users to override the system prompt so i think `use_default_system` is not required.



> why is this needed? To use the default system prompt, you can just leave the content empty.

Sorry for late, some template like [qwen2](https://modelscope.cn/models/qwen/Qwen2-72B-Instruct/file/view/master?fileName=tokenizer_config.json&status=1) enforce a predefined system prompt. Even when given customized system prompt.  But the logic in [chat_template.py](https://github.com/sgl-project/sglang/blob/main/python/sglang/lang/chat_template.py#L45) seems to prioritize customized system prompt over default. 

Hence I use `default system prompt` to control whether enforce use system prompt.

Reply:
I see. But i still prefer the flexibility of allowing users to forcibly change the system prompt if they want. For example, the developer of Qwen might still want to compare different system prompts. With your change, it is impossible to do this. Flexibility is favored here. Could you revert the change of `use_default_system`?
The other two chat templates look good!